web/frontend/src/components/slam.vue

```diff
@@ -56,17 +56,14 @@ const fetchSLAMMap = (name: string): Promise<Uint8Array> => {
       const chunk = res.getPointCloudPcdChunk_asU8();
       chunks.push(chunk);
     });
-    getPointCloudMap.on('status', (status: { code: number, details: string, metadata: string }) => {
-      if (status.code !== 0) {
-        const error = {
-          message: status.details,
-          code: status.code,
-          metadata: status.metadata,
-        };
+    getPointCloudMap.on('status', (status?) => {
+      const { code, details, metadata } = status!;
+      if (code !== 0) {
+        const error = { message: details, code, metadata };
         reject(error);
       }
     });
-    getPointCloudMap.on('end', (end: { code: number }) => {
+    getPointCloudMap.on('end', (end?: { code: number }) => {
```

Comment:
1. what does it mean if `end` is undefined?
2. why is this changing?

Reply:
> 1. what does it mean if end is undefined?

This means the response was closed without a valid grpc-status. This can happen if the server shuts down or receives a malformed request.

> 2. why is this changing?

My best guess is that types were mis-configured on the typescript side - we recently made [a change to address this](https://github.com/viamrobotics/viam-typescript-sdk/pull/56). In this case, we were properly handing status being undefined, but just did not have the correct type signature to reflect this.